### ISSUE TITLE
Uncaught `Premature close` stream error on Node.js v13.5.0

### ISSUE THREAD

**Issue by szmarczak** [sindresorhus/got#1003]
From https://travis-ci.org/sindresorhus/got/jobs/627203773

```
...
  Uncaught exception in test/stream.ts
  /home/travis/build/sindresorhus/got/node_modules/@szmarczak/http-timer/source/index.ts:75
  Error [ERR_STREAM_PREMATURE_CLOSE]: Premature close
...
  1 test failed
  1 test skipped
  1 uncaught exception
  gzip.ts › handles gzip error - stream
  /home/travis/build/sindresorhus/got/test/gzip.ts:51
   50:                                                   
   51:   await t.throwsAsync(getStream(got.stream('')), {
   52:     name: 'ReadError',                            
  Promise rejected with unexpected exception:
  Error {
    bufferedData: '',
    code: 'Z_DATA_ERROR',
    errno: -3,
    message: 'incorrect header check',
  }
  Expected name to equal:
  'ReadError'
```

**Comment by szmarczak** [comment]
Ah, forgot about #990

**Comment by szmarczak** [comment]
Actually this is still happening somehow: https://travis-ci.org/sindresorhus/got/jobs/627255765

**Comment by szmarczak** [comment]
Indeed. Something in Node.js core changed. I upgraded my local Node.js from v13.0.1 to v13.5.0 and it fails on my side too.

**Comment by szmarczak** [comment]
Spent an hour looking for a bug in Got. Nice surprise: https://github.com/nodejs/node/commit/648088289d619bfb149fe90316ce0127083c4c99

**Comment by szmarczak** [comment]
See https://github.com/nodejs/node/issues/31029

**Comment by lpinca** [comment]
See https://github.com/nodejs/node/commit/648088289d619bfb149fe90316ce0127083c4c99.

**Comment by szmarczak** [comment]
Fixed in v13.6.0
